com.unity.perception/Runtime/GroundTruth/Labelers/Visualization/HUDPanel.cs

```diff
@@ -110,7 +113,10 @@ internal void OnDrawGUI()
                 {
                     GUILayout.BeginHorizontal();
                     GUILayout.Space(5);
-                    GUILayout.Label(entry.Key);
+                    var k = new StringBuilder(entry.Key.Substring(0, Math.Min(entry.Key.Length, k_MaxKeyLength)));
+                    if (k.Length != entry.Key.Length) 
```

Comment:
Minor issue - this approach will potentially turn "A long string" into "A long strin..." - increasing the string length instead of shortening it.

Reply:
Yeah, I guess it could be handled a little better, where we remove 3 letters off the back if shortening. The main thing I was trying to do was make sure that we didn't go pass the buffer, so that is taking into account the size of the ellipsis.